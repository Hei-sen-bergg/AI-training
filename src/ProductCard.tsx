import React from 'react';

interface ProductCardProps {
  name: string;
  price: number;
  image: string;
}

const ProductCard: React.FC<ProductCardProps> = ({ name, price, image }) => {
  return (
    <div className="bg-white rounded-lg shadow-md overflow-hidden w-64 m-4 flex flex-col items-center">
      <img src={image} alt={name} className="w-full h-40 object-cover" />
      <div className="p-4 w-full flex flex-col items-center">
        <h2 className="text-lg font-semibold mb-2 text-gray-800">{name}</h2>
        <p className="text-xl font-bold text-blue-600">${price.toFixed(2)}</p>
      </div>
    </div>
  );
};

export default ProductCard;
