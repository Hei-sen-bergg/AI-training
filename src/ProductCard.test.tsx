// Write Vitest + React Testing Library tests for the ProductCard component.
// - It should render product name, price, and image correctly
// - It should display formatted price with "$"
// - It should match snapshot


import { render, screen } from '@testing-library/react';
import ProductCard from './ProductCard';
import { describe, it, expect } from 'vitest';
import '@testing-library/jest-dom';

describe('ProductCard', () => {
  const name = 'Test Product';
  const price = 19.99;
  const image = 'test-image.jpg';

  it('renders product name, price, and image correctly', () => {
    render(<ProductCard name={name} price={price} image={image} />);
    expect(screen.getByText(name)).toBeInTheDocument();
    expect(screen.getByText('$19.99')).toBeInTheDocument();
    const img = screen.getByRole('img');
    expect(img).toHaveAttribute('src', image);
    expect(img).toHaveAttribute('alt', name);
  });

  it('displays formatted price with "$"', () => {
    render(<ProductCard name={name} price={price} image={image} />);
    expect(screen.getByText('$19.99')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<ProductCard name={name} price={price} image={image} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

