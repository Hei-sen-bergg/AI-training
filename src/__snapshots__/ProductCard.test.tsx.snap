// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-lg shadow-md overflow-hidden w-64 m-4 flex flex-col items-center"
  >
    <img
      alt="Test Product"
      class="w-full h-40 object-cover"
      src="test-image.jpg"
    />
    <div
      class="p-4 w-full flex flex-col items-center"
    >
      <h2
        class="text-lg font-semibold mb-2 text-gray-800"
      >
        Test Product
      </h2>
      <p
        class="text-xl font-bold text-blue-600"
      >
        $19.99
      </p>
    </div>
  </div>
</DocumentFragment>
`;
